lib/system_win32.c

```diff
@@ -97,8 +97,7 @@ CURLcode Curl_win32_init(long flags)
   if(s_hIpHlpApiDll) {
     /* Get the address of the if_nametoindex function */
     IF_NAMETOINDEX_FN pIfNameToIndex =
-      CURLX_FUNCTION_CAST(IF_NAMETOINDEX_FN,
-                          (GetProcAddress(s_hIpHlpApiDll, "if_nametoindex")));
+      (IF_NAMETOINDEX_FN)GetProcAddress(s_hIpHlpApiDll, "if_nametoindex"));
```

Comment:
```suggestion
      (IF_NAMETOINDEX_FN)GetProcAddress(s_hIpHlpApiDll, "if_nametoindex");
```

I'm pretty sure we needed CURLX_FUNCTION_CAST for CI, but I guess we'll see.

Reply:
why do we need it? a "manual" cast should do the same, right?

Reply:
Yeah. But I think we had problems that required it. See 25d2a1bae78 for example

/cc @MarcelRaad

Reply:
Yes, a manual cast works too, but the intermediate `(void (*) (void)` cast is important to silence the `-Wcast-function-type` GCC warning.

Reply:
Ok, thanks. I'll experiment a little further and see where it might go.